package.json

```
@@ -22,10 +22,17 @@
   "exports": {
     ".": {
       "import": "./lib/index.mjs",
-      "require": "./lib/index.js"
+      "require": "./lib/index.js",
+      "types": "./types/index.d.ts"
+    },
+    "./lib/*": {
+      "default": "./lib/*.js",
+      "types": "./types/*.d.ts"
```

```suggestion
      "types": "./types/*"
```
This would probably also need to be without the extension as mentioned in #14610

---

Here the extension must be specified because `default` specifies it (but I need to run a battery of tests on this PR to be 100% sure)